src/utils.jl

```diff
@@ -68,6 +68,143 @@ converted to a `Matrix` before computing the log.
+                        "All eigenvalues must be positive to compute a real logarithm.",
+                    ),
+                )
+            end
+            mul!(Y, eigenF.vectors .* log.(eigenF.values'), eigenF.vectors')
+        elseif istriu(A)
+            i = findfirst(≤(0), @view(A[diagind(A)]))
+            if i !== nothing
+                throw(
+                    DomainError(
+                        A[i, i],
+                        "All eigenvalues must be positive to compute a real logarithm.",
+                    ),
+                )
+            end
+            copyto!(Y, real(log(UpperTriangular(A))))
+        else
+            schurF = schur(convert(Matrix, real(A)))
+            i = findfirst(x -> isreal(x) && real(x) ≤ 0, schurF.values)
+            if i !== nothing
+                throw(
+                    DomainError(
+                        schurF.values[i],
+                        "All eigenvalues must be positive to compute a real logarithm.",
+                    ),
+                )
+            end
+            if istriu(schurF.T)
+                mul!(Y, schurF.Z, real(log(UpperTriangular(schurF.T))) * schurF.Z')
+            else
+                schurS = schur(complex(schurF.T))
+                Y .= real.(schurS.Z * log(UpperTriangular(schurS.T)) * schurS.Z')
+                mul!(Y, schurF.Z * Y, schurF.Z')
+            end
+        end
+    else
+        copyto!(Y, log_safe(A))
+    end
+    return Y
+end
+
+"""
+    mul!_safe(Y, A, B) -> Y
+
+Call `mul!` safely, that is, `A` and/or `B` are permitted to alias with `Y`.
+"""
+mul!_safe(Y, A, B) = (Y === A || Y === B) ? copyto!(Y, A * B) : mul!(Y, A, B)
+
+@doc raw"""
+    realify(X::AbstractMatrix{T𝔽}, 𝔽::AbstractNumbers) -> Y::AbstractMatrix{<:Real}
+
+Given a matrix $X ∈ 𝔽^{n × n}$, compute $Y ∈ ℝ^{m × m}$, where $m = n \operatorname{dim}_𝔽$,
+and $\operatorname{dim}_𝔽$ is the [`real_dimension`](@ref) of the number field $𝔽$, using
+the map $ϕ \colon X ↦ Y$, that preserves the matrix product, so that for all
+$C,D ∈ 𝔽^{n × n}$,
+````math
+ϕ(C) ϕ(D) = ϕ(CD).
+````
+See [`realify!`](@ref) for an in-place version, and [`unrealify!`](@ref) to compute the
+inverse of $ϕ$.
+"""
+function realify(X, 𝔽)
+    n = LinearAlgebra.checksquare(X)
+    nℝ = real_dimension(𝔽) * n
+    Y = allocate(X, real(eltype(X)), nℝ, nℝ)
+    return realify!(Y, X, 𝔽, n)
+end
+realify(X, ::typeof(ℝ)) = X
+
+"""
+    realify!(Y::AbstractMatrix{<:Real}, X::AbstractMatrix{T𝔽}, 𝔽::AbstractNumbers)
+
+In-place version of [`realify`](@ref).
+"""
+realify!(Y, X, 𝔽)
+
+@doc raw"""
+    realify!(Y::AbstractMatrix{<:Real}, X::AbstractMatrix{<:Complex}, ::typeof(ℂ))
+
+Given a complex matrix $X = A + iB ∈ ℂ^{n × n}$, compute its realified matrix
+$Y ∈ ℝ^{2n × 2n}$, written
+where
+````math
+Y = \begin{pmatrix}A & -B \\ B & A \end{pmatrix}.
+````
+"""
+function realify!(Y, X, ::typeof(ℂ), n=LinearAlgebra.checksquare(X))
+    axul, axlr = 1:n, (n + 1):(2n)
+    @views begin
```

Comment:
Oh cool, this is marked as not covered because when expanded, the `@views` is a no-op:
```julia
julia> @macroexpand @views begin
               Y[axul, axul] .= Y[axlr, axlr] .= real.(X)
               Y[axlr, axul] .= imag.(X)
               Y[axul, axlr] .= .-imag.(X)
           end
quote
    #= REPL[153]:2 =#
    Y[axul, axul] .= (Y[axlr, axlr] .= real.(X))
    #= REPL[153]:3 =#
    Y[axlr, axul] .= imag.(X)
    #= REPL[153]:4 =#
    Y[axul, axlr] .= (.-)(imag.(X))
end
```